### PR TITLE
btf: Add bindings for BTF_KIND_FLOAT

### PR DESCRIPTION
**Pull request by dave-tucker**
Fixes: #34

Signed-off-by: Dave Tucker <dave@dtucker.co.uk>

### CODE REVIEW

**Comment by alessandrod** [comment]
Thanks for this Dave! Planning to compile a newer and test later today.

**Comment by dave-tucker** [comment]
@alessandrod I've expanded this PR a little.
1. Added some test cases for parsing the various BTF types using data from `/sys/kernel/btf/vmlinux` on my system.
2. Made `Btf::parse` issue an error if we run in to an Unknown btf type, as that was what caused the infinite loop.

That last commit got quite big since the `BtfType::Unknown` variant was then never constructed... and therefore a whole lot of tunctions that were returning `Option` didn't need to any more.

File: aya/src/obj/btf/btf.rs [hunk 1]
```diff
@@ -143,7 +143,7 @@ impl Btf {
         }
 
         let mut data = &data[base..base + type_len];
-        let mut types = vec![BtfType::Unknown];
+        let mut types = vec![];
         while !data.is_empty() {
```
Comment (alessandrod) [inline aya/src/obj/btf/btf.rs:147]: The first type must always be `Unknown` or it will break `type_by_id`

File: aya/src/obj/btf/types.rs [hunk 2]
```diff
@@ -150,10 +154,9 @@ impl BtfType {
 
         use BtfType::*;
         match self {
-            Unknown => 0,
-            Fwd(_) | Const(_) | Volatile(_) | Restrict(_) | Ptr(_) | Typedef(_) | Func(_) => {
-                ty_size
-            }
+            Unknown => ty_size,
```
Comment (alessandrod) [inline aya/src/obj/btf/types.rs:157]: As discussed on discord, to fix the infinite loop on `Unknown` bug, this should be `Unknown => ty_size`

**Comment by alessandrod** [comment]
Good stuff! See the comments then it's good to go

**Review by alessandrod — CHANGES_REQUESTED** [review]

File: aya/src/obj/btf/btf.rs (hunk 1 above)
Comment (dave-tucker) [reply aya/src/obj/btf/btf.rs]: fixed!

File: aya/src/obj/btf/types.rs (hunk 2 above)
Comment (dave-tucker) [reply aya/src/obj/btf/types.rs:157]: fixed in https://github.com/alessandrod/aya/pull/35/commits/4465131199e36f288c552a9cbc609fa6720a3d2f

**Review by alessandrod — APPROVED** [review]

**Pull request merged**